pkg/datarepair/checker/checker.go

```diff
@@ -99,14 +102,31 @@ func (c *checker) identifyInjuredSegments(ctx context.Context) (err error) {
 					return Error.New("error getting offline nodes %s", err)
 				}
 				numHealthy := len(nodeIDs) - len(missingPieces)
-				if int32(numHealthy) < pointer.Remote.Redundancy.RepairThreshold {
+				if (int32(numHealthy) >= pointer.Remote.Redundancy.MinReq) && (int32(numHealthy) < pointer.Remote.Redundancy.RepairThreshold) {
 					err = c.repairQueue.Enqueue(&pb.InjuredSegment{
 						Path:       string(item.Key),
 						LostPieces: missingPieces,
 					})
 					if err != nil {
 						return Error.New("error adding injured segment to queue %s", err)
 					}
+				} else if int32(numHealthy) < pointer.Remote.Redundancy.MinReq {
+					// make an entry in to the irreparable table
+					rmtSegInfo := &irreparabledb.RemoteSegmentInfo{
+						SegmentPath:        item.Key,
+						SegmentDetail:      item.Value,
+						LostPiecesCount:    int64(len(missingPieces)),
+						RepairUnixSec:      time.Now().Unix(),
+						RepairAttemptCount: int64(1),
+					}
+
+					//check if that entry already exists
+					dbrmtSegInfo, err := c.irrdb.Get(ctx, rmtSegInfo.SegmentPath)
```

Comment:
This is still not safe. It should only make one call to the database.

```
err := c.irrdb.IncrementAttemptCount(ctx, segmentInfo)
```

Reply:
```
server 1                                      server 2

segment, err := db.Get(path)
                                              segment, err := db.Get(path)
if err == nil {                               if err == nil {
                                                  return db.Create(segmentInfo)
    return db.Create(segmentInfo)
    ^^^ this will return an error, because info already exists
```

Reply:
you right. I was under the impression that there is a satellite and only once instance of repair queue and one instance of irreparable are running. It could potentially case issues. Changed the code.